### PR TITLE
What: #1480 Add tech leadership principles guidance

### PR DESCRIPTION
**Pull request by TheFoxAtWork**
Why:
 * CoCC is receiving reports of violations they are not chartered to manage

This change address the need by:
 * creates the guidance file with enforcement/escalation information

 What is missing:
 * We need the email alias set up for escalation, CC @mrbobbytables

### CODE REVIEW

**Comment by dims** [comment]
thanks for writing this up @TheFoxAtWork 

https://github.com/kubernetes/steering/blob/main/charter.md#removal is an example of a documented process in k8s.

File: docs/tech_leadership_principles_guidance.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+When an individual's behavior fails to embody our principles and values and/or their personal/private views are exposed to the community, it has the potential to subversively abuse that imbalance and negatively harm the project and its community. A leader’s behavior and views has the singular ability to build or break our projects and our community.
+
+### Prevention
+
+All projects, groups, and committees are expected to establish a policy or governance process that allows for the removal of a leader for failure to continually align with our principles and values through a vote of no confidence after attempts to constructively correct (restorative justice) have occurred. It is unrealistic to detail every possible offense or behavior that could warrant such action and each project may have additional considerations to evaluate as part of determining the need for a vote of no confidence.
```
Comment (leonardpahlke) [inline docs/tech_leadership_principles_guidance.md:45]: All projects, groups and committees are expected to establish …

Was looking for a resource, found this one: https://github.com/cncf/project-template/blob/main/CONTRIBUTOR_LADDER.md#involuntary-removal-or-demotion

May be good to link it. And Or mention tag contrib strat governance reviews for help.

File: docs/tech_leadership_principles_guidance.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Additional Information and Guidance
+
+### Conflict is likely
+
+Innovation is often born from the diversity of opinion, perspective, and experience of individuals collaborating towards a common goal or objective. Fostering an engaging and exploratory discussion, debate, and exchange of ideas to drive innovation requires a foundation of respect, grace, and curiosity so all voices may be elevated, heard, and incorporated into the discussion.
+
+Any personal, social views and political opinions exposed to the community can destroy and undermine trust of the leader holding them because those personal views may not align with CNCF values and may be seen to promote intolerance. Individuals have a right to their personal, private perspectives, but as leaders in our community we are expected to embody and practice the technical leadership principles and uphold our Code of Conduct through our behavior and engagements, serving as the arbiter of equity and fairness within projects, within the community, within CNCF spaces, and at CNCF events.
```
Comment (leonardpahlke) [inline docs/tech_leadership_principles_guidance.md:35]: This paragraph contains information that's also in the next one (about trust). Maybe move and merge? I like the subsections. (minor comment, just smth i noticed)

File: docs/tech_leadership_principles_guidance.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Escalating to the TOC
+
+In the event that the project or group cannot resolve this conflict with the resources available to them, the project should escalate to the TOC through the private mailing list where the TOC Chair, Vice Chair, or CNCF Staff who will triage.
+
+To escalate to the TOC, please email: cncf-private-toc@lists.cncf.io 
+
+If the escalation concerns a current TOC member, please contact the TOC Chair, Vice Chair, and/or CNCF Staff directly.
+
+Once escalated, two TOC members will be assigned to guide and mentor the project or group and its leadership on methods and options to arrive at an agreeable resolution. Resolutions are crafted to allow for the best outcome for the project or group and its future. The TOC may suggest changes among the leadership roles, responsibilities, and project governance to support that outcome.
```
Comment (leonardpahlke) [inline docs/tech_leadership_principles_guidance.md:27]: 2 TOC members will be assigned by the TOC Chair? Probably, right

Comment (leonardpahlke) [reply docs/tech_leadership_principles_guidance.md:27]: The TOC may suggest XYZ. (Obvious question) In the unlikely event that suggestions do not help? CNCF TOC / GB can vote on a project archival or similar? Enforce CoC. Maybe it's good to keep this unanswered.

**Review by dims — APPROVED** [review]
LGTM

**Review by leonardpahlke — APPROVED** [review]
LGTM

**Review by angellk — APPROVED** [review]
lgtm

**Pull request merged**